### DISCUSSION TITLE
Undefined Reference To udev_device_get_devnode

### DISCUSSION THREAD

**Discussion by ThothK** [libusb/libusb#1565] [Q&A] [answered]
Hi,

I am getting linker errors specifying undefined references to functions in udev. I have installed libudev-dev - libudev development files. An example of the error:
/usr/bin/ld: /Libraries/USB1/Debian_x64_Debug_Static_GNU/libusb-1.0.a(linux_udev.o): in function `udev_device_info':
/Source/USB_Lib/libusb_1_0_25/libusb/os/linux_udev.c:214: undefined reference to `udev_device_get_devnode'
/usr/bin/ld: /Source/USB_Lib/libusb_1_0_25/libusb/os/linux_udev.c:219: undefined reference to `udev_device_get_sysname'

Some of the other functions that cannot be found include:
/usr/bin/ld: /Source/USB_Lib/libusb_1_0_25/libusb/os/linux_udev.c:55: undefined reference to `udev_monitor_new_from_netlink'
/usr/bin/ld: /Source/USB_Lib/libusb_1_0_25/libusb/os/linux_udev.c:61: undefined reference to `udev_monitor_filter_add_match_subsystem_devtype'
/usr/bin/ld: /Source/USB_Lib/libusb_1_0_25/libusb/os/linux_udev.c:67: undefined reference to `udev_monitor_enable_receiving'
/usr/bin/ld: /Source/USB_Lib/libusb_1_0_25/libusb/os/linux_udev.c:72: undefined reference to `udev_monitor_get_fd'
/usr/bin/ld: /Source/USB_Lib/libusb_1_0_25/libusb/os/linux_udev.c:121: undefined reference to `udev_monitor_unref'
/usr/bin/ld: /Source/USB_Lib/libusb_1_0_25/libusb/os/linux_udev.c:125: undefined reference to `udev_unref'

If anyone has any suggestions about what I should install please let me know.

**Comment by tormodvolden** [comment]
It says Debian, but it doesn't look like Debian. You must provide context and details, exact command line that is executed etc. See https://github.com/libusb/libusb/wiki/Troubleshooting

**Comment by ThothK** [comment, marked as answer]
Hi People,

I am closing this. The problem was the version of the USB library I was using. Once I straightened that out, the linker errors went away.
